### DISCUSSION TITLE
Large dynamic text isn't page breaked in PDF

### DISCUSSION THREAD

**Discussion by LeonardoIPM** [eclipse-birt/birt#1892] [Q&A] [answered]
We're trying to create a page numeration using Global and Local variables (We can't use AutoText option because we can't count the first page).
The issue is that, when the dynamic text has the size of 2 or more pages, it is considered as just one page. Scripts in onPageBreak/onPageEnd/onRender don't trigger and I can't create a counter to the pages.
I don't know how AutoText Page works, but it seems to be the only element that can count PDF pages correctly, but I don't know if is there a way to manipulate this value, or to emulate the same process".

I believe this is a known behavior, is there a workaround for this?

Simple example creating a counter in onPageEnd script and setting the value on text box: 
[TestDynamic.zip](https://github.com/user-attachments/files/16872105/TestDynamic.zip)

**Comment by speckyspooky** [comment, marked as answer]
This is a kind of standard topic which is currently not implemented for all cases.
Because the topic is that the elements "text", "dynamice text" and "data elements" can have dynamic long text
and therefore there is no solution to calculate the full length over longt text elements. The current calculation absed on the height of elements
to get the page count.

Yes, there are different results of page break if you use HTML or PDF due to the output handling.
But currently there is no direct solution given at the time.

**Reply by LeonardoIPM** [reply]
The AutoText works very well, but I believe it is generated calculating the height of the document, not the elements right? Would be possible to create simple ways to manage this variable? If we could apply a -1 to exclude the sumary, would resolve our issue for example.
If it isn't the best approach, never mind.
Thanks for your time!

**Reply by speckyspooky** [reply]
Of course the height of the elements based on designer height will be used for the calculation and not the content result.
It must be investigated to see wich ways are possible but I think it isn't possible without source changes.
